### PR TITLE
Typed style obj

### LINKED ISSUE #43: type style obj
We can use the [@bs.obj] external.

### PR DESCRIPTION
**Pull request by glennsl**
(Partially) fixes #43 

Should there be more ways of manipulating style objects? How about accessing style objects? Should there be accessor functions?

### CODE REVIEW

**Comment by sgrove** [comment]
How does this work with vendor-prefixed properties? I noticed `WebkitUserSelect` was changed to `userSelect`, but isn't there a danger that designers will be slowed down if they have to wait for an update to the style type to use a new style?

**Comment by chenglou** [comment]
@sgrove we can always just add `WebkitUserSelect` to our list.

I'm hesitating here a bit. It looks a bit less nice than the current free-form style obj. But hummm type checking...

**Comment by glennsl** [comment]
We can have both. The simplest way would be having a function `withCustomProp : style => string => string => style = ...`. A bit cumbersome to use, but maybe it should be.

**Comment by sgrove** [comment]
@chenglou I meant the time delta between designer-tries-to-use-property-they're-accustomed-to-that's-missing and property-is-added-to-list-new-release-is-pushed-deps-are-updated, and that's assuming the designer doesn't just feel frustrated and doesn't submit the property (or doesn't know how!)

File: src/reactDOMRe.re [hunk 1]
```diff
@@ -175,3 +177,412 @@ external props :
...
+     * Motion Path Level 1 - FPWD
+     * Color Level 4 - FPWD
+     * SVG Strokes - FPWD
+     * Table Level 3 - FPWD
+     */
+
+    unit =>
+    style = "" [@@bs.obj];
+
+    let combine : style => style => style = [%bs.raw {|
```
Comment (chenglou) [inline src/reactDOMRe.re:582]: Ahh, this is kinda unfortunate

**Comment by chenglou** [comment]
About time!

**Pull request merged**